src/uucore/src/lib/features/process.rs

```diff
@@ -107,50 +140,180 @@ impl ChildExt for Child {
+                Ok(_) => {
+                    if start.elapsed() >= timeout && !timeout.is_zero() {
+                        return Ok(WaitOrTimeoutRet::TimedOut);
+                    }
+                    // The set is modified to contain the readable ones;
+                    // if empty, we'd stall on the read. However, this may
+                    // happen spuriously, so we try to select again.
+                    if fd_set.contains(self_pipe.0.as_fd()) {
+                        let mut buf = [0];
+                        self_pipe.0.read_exact(&mut buf)?;
+                        return match buf[0] {
+                            // SIGCHLD
+                            1 => match self.try_wait()? {
+                                Some(e) => Ok(WaitOrTimeoutRet::InTime(e)),
+                                None => Ok(WaitOrTimeoutRet::InTime(ExitStatus::default())),
+                            },
+                            // Custom signals on zero timeout still succeed.
+                            2 if timeout.is_zero() => {
+                                Ok(WaitOrTimeoutRet::InTime(ExitStatus::default()))
+                            }
+                            // We received a custom signal and fail.
+                            2 => Ok(WaitOrTimeoutRet::CustomSignaled),
+                            _ => unreachable!(),
+                        };
+                    }
+                }
             }
+        }
+    }
+}
 
-            // XXX: this is kinda gross, but it's cleaner than starting a thread just to wait
-            //      (which was the previous solution).  We might want to use a different duration
-            //      here as well
-            thread::sleep(Duration::from_millis(100));
+#[cfg(feature = "pipes")]
+#[allow(clippy::unnecessary_fallible_conversions, clippy::useless_conversion)]
+fn duration_to_timeval_elapsed(time: Duration, start: Instant) -> Option<TimeVal> {
+    if time.is_zero() {
+        None
+    } else {
+        let elapsed = start.elapsed();
+        // This code ensures we do not overflow on any platform.
+        let mut val = TimeVal::new(
+            time.as_secs()
+                .try_into()
+                .unwrap_or(libc::time_t::MAX)
+                .saturating_sub_unsigned(elapsed.as_secs().try_into().unwrap_or_default()),
+            time.subsec_micros()
+                .saturating_sub(elapsed.subsec_micros().try_into().unwrap_or_default())
+                .try_into()
+                .unwrap_or_default(),
+        );
+        // Not all platforms support uncapped times (read: macOS). However,
+        // POSIX has left it unspecified so we will just apply it globally,
+        // except on Linux, to be safe; it's the most portable option.
+        // https://pubs.opengroup.org/onlinepubs/007904875/basedefs/sys/types.h.html#tag_13_67
+        if val.tv_sec() >= 100_000_000 && !cfg!(target_os = "linux") {
```

Comment:
please move 100_000_000 into a variable with a nice name :)
feel like a magic number here

Reply:
Sure, gimme 5 mins to power on the laptop :)